### PR TITLE
[WIP] Implementation of native code wrapper for Cordova

### PR DESCRIPTION
**Pull request by josemigallas**
## Motivation
https://issues.jboss.org/browse/AGCORDOVA-199

## Description
TODO

## Progress
- [x] Plugin definition
- [x] Integration with example app
- [ ] Android bits
- [ ] iOS bits
- [ ] Integration with @aerogearservices/core

## Additional Notes
Do not merge, work in progress.

### CODE REVIEW

File: packages/core-cordova/src/android/MobileCoreModule.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    return false;
+  }
+
+  @Override
+  public String getServiceName() {
+    return "MobileCore";
+  }
+
+  public void getDeviceMetrics(CallbackContext callbackContext) {
+    // try {
```
Comment (wtrocki) [inline packages/core-cordova/src/android/MobileCoreModule.java]: Copy paste Android code.

File: packages/core-cordova/src/android/MobileCoreModule.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    //   promise.resolve(deviceMetrics);
+
+    // } catch (Exception e) {
+    //   promise.reject("GetDeviceMetricsError", e.getMessage());
+    // }
+    callbackContext.success("Device metrics");
+  }
+
+  public void getAppMetrics(CallbackContext callbackContext) {
+    // try {
```
Comment (wtrocki) [inline packages/core-cordova/src/android/MobileCoreModule.java]: Same as above (we can think about reusability later)

File: example/cordova/package.json [hunk 3]
```diff
@@ -5,33 +5,33 @@
...
     "browserify": "node node_modules/browserify/bin/cmd.js www/js/index.js -o www/js/bundle.js",
     "copy-f7": "cpy node_modules/framework7/dist/js/*.* www/framework7/js && cpy node_modules/framework7/dist/css/*.* www/framework7/css && cpy node_modules/framework7-icons/fonts/*.* www/fonts && cpy node_modules/framework7-icons/css/framework7-icons.css www/css && cpy node_modules/material-design-icons/iconfont/*.{eot,ttf,woff,woff2} www/fonts && cpy node_modules/material-design-icons/iconfont/material-icons.css www/fonts",
-    "postinstall": "npm run copy-f7"
+    "run:android:clean": "cordova clean && rm -rf node_modules && npm install && cordova run android"
   },
-  "author": "Apache Cordova Team",
   "license": "Apache-2.0",
   "dependencies": {
-    "@aerogearservices/core": "file:../../packages/core",
+    "@aerogearservices/core-cordova": "file:../../packages/core-cordova",
```
Comment (pb82) [inline example/cordova/package.json]: why is it different from aerogear/core?

File: example/cordova/config.xml [hunk 4]
```diff
@@ -1,9 +1,7 @@
 <?xml version='1.0' encoding='utf-8'?>
-<widget id="org.aerogear.services.cordova.example" version="1.0.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
+<widget id="org.aerogear.mobile.example.cordova" version="1.0.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
```
Comment (wtrocki) [inline example/cordova/config.xml]: Please refrain from changing names, packages etc.

File: packages/core-cordova/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Aerogear Mobile SDK for Cordova
```
Comment (wtrocki) [inline packages/core-cordova/README.md]: Going to document this in my ticket. it's fine to have TODO

**Pull request merged**